pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -250,14 +250,23 @@ cdef class _Timestamp(datetime):
         return np.datetime64(self.value, 'ns')
 
     def __add__(self, other):
-        cdef int64_t other_int, nanos
+        cdef:
+            int64_t other_int, nanos
 
         if is_timedelta64_object(other):
             other_int = other.astype('timedelta64[ns]').view('i8')
             return Timestamp(self.value + other_int,
                              tz=self.tzinfo, freq=self.freq)
 
         elif is_integer_object(other):
+            if self.freq is None:
```

Comment:
is it necessary for freq to be none here? (we don't have this check elsewhere)

Reply:
Should be not-None, good catch.

Reply:
does this need to be changed?